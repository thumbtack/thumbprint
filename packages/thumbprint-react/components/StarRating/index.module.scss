// Each image is a single gray star on top of a single yellow star.

$small-spacing-adjustment: -1px;
$small-height: 12px;
$small-width: 70px;
$small-image: url('data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTQiIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAxNCAyNCIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPHBhdGggZD0iTTcuMzQyMTUgMC4yNTc2MzFMOC42MjcyNSA0LjI0NTdMMTIuNjQyNCA0LjI5NDQzQzEyLjk4OCA0LjI5ODY2IDEzLjEzMSA0Ljc2MDE5IDEyLjg1MzkgNC45NzYxNEw5LjYzMjUgNy40ODkzNUwxMC44MjkgMTEuNTA3NEMxMC45MzE4IDExLjg1MzIgMTAuNTU3NyAxMi4xMzggMTAuMjc1NyAxMS45Mjg2TDcuMDAwMiA5LjQ5MzY2TDMuNzI0MyAxMS45MjgyQzMuNDQyNyAxMi4xMzc2IDMuMDY4MTMgMTEuODUyMiAzLjE3MDkzIDExLjUwN0w0LjM2NzUgNy40ODg4OEwxLjE0NjExIDQuOTc1NjdDMC44Njg5ODQgNC43NTk3MiAxLjAxMjAyIDQuMjk4MTkgMS4zNTc1MyA0LjI5Mzk2TDUuMzcyNzUgNC4yNDUyM0w2LjY1Nzg1IDAuMjU3MTYxQzYuNzY4NyAtMC4wODU3OTg2IDcuMjMxNzUgLTAuMDg1Nzk4NiA3LjM0MjE1IDAuMjU3NjMxWiIgZmlsbD0iI0QzRDRENSIvPgo8cGF0aCBkPSJNNy4zNDIxNSAxMi4yNTc2TDguNjI3MjUgMTYuMjQ1N0wxMi42NDI0IDE2LjI5NDRDMTIuOTg4IDE2LjI5ODcgMTMuMTMxIDE2Ljc2MDIgMTIuODUzOSAxNi45NzYxTDkuNjMyNSAxOS40ODk0TDEwLjgyOSAyMy41MDc0QzEwLjkzMTggMjMuODUzMiAxMC41NTc3IDI0LjEzOCAxMC4yNzU3IDIzLjkyODZMNy4wMDAyIDIxLjQ5MzdMMy43MjQzIDIzLjkyODJDMy40NDI3IDI0LjEzNzYgMy4wNjgxMyAyMy44NTIyIDMuMTcwOTMgMjMuNTA3TDQuMzY3NSAxOS40ODg5TDEuMTQ2MTEgMTYuOTc1N0MwLjg2ODk4NCAxNi43NTk3IDEuMDEyMDIgMTYuMjk4MiAxLjM1NzUzIDE2LjI5NEw1LjM3Mjc1IDE2LjI0NTJMNi42NTc4NSAxMi4yNTcyQzYuNzY4NyAxMS45MTQyIDcuMjMxNzUgMTEuOTE0MiA3LjM0MjE1IDEyLjI1NzZaIiBmaWxsPSIjMkRCNzgzIi8+Cjwvc3ZnPgo=');

$medium-spacing-adjustment: -2px;
$medium-height: 18px;
$medium-width: 110px;
$medium-image: url('data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjIiIGhlaWdodD0iMzYiIHZpZXdCb3g9IjAgMCAyMiAzNiIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPHBhdGggZD0iTTExLjUxMzIgMTguMzg2NEwxMy40NDA5IDI0LjM2ODZMMTkuNDYzNyAyNC40NDE3QzE5Ljk4MiAyNC40NDggMjAuMTk2NSAyNS4xNDAzIDE5Ljc4MDggMjUuNDY0MkwxNC45NDg3IDI5LjIzNEwxNi43NDM2IDM1LjI2MTFDMTYuODk3OCAzNS43Nzk4IDE2LjMzNjYgMzYuMjA3MSAxNS45MTM1IDM1Ljg5MjlMMTEuMDAwMyAzMi4yNDA1TDYuMDg2NDUgMzUuODkyMkM1LjY2NDA1IDM2LjIwNjQgNS4xMDIxOSAzNS43Nzg0IDUuMjU2NCAzNS4yNjA0TDcuMDUxMjYgMjkuMjMzM0wyLjIxOTE3IDI1LjQ2MzVDMS44MDM0OCAyNS4xMzk2IDIuMDE4MDMgMjQuNDQ3MyAyLjUzNjMgMjQuNDQwOUw4LjU1OTEyIDI0LjM2NzlMMTAuNDg2OCAxOC4zODU3QzEwLjY1MyAxNy44NzEzIDExLjM0NzYgMTcuODcxMyAxMS41MTMyIDE4LjM4NjRaIiBmaWxsPSIjMkRCNzgzIi8+CjxwYXRoIGQ9Ik0xMS41MTMyIDAuMzg2NDQ2TDEzLjQ0MDkgNi4zNjg1NkwxOS40NjM3IDYuNDQxNjVDMTkuOTgyIDYuNDQ3OTkgMjAuMTk2NSA3LjE0MDI4IDE5Ljc4MDggNy40NjQyMUwxNC45NDg3IDExLjIzNEwxNi43NDM2IDE3LjI2MTFDMTYuODk3OCAxNy43Nzk4IDE2LjMzNjYgMTguMjA3MSAxNS45MTM1IDE3Ljg5MjlMMTEuMDAwMyAxNC4yNDA1TDYuMDg2NDUgMTcuODkyMkM1LjY2NDA1IDE4LjIwNjQgNS4xMDIxOSAxNy43Nzg0IDUuMjU2NCAxNy4yNjA0TDcuMDUxMjYgMTEuMjMzM0wyLjIxOTE3IDcuNDYzNUMxLjgwMzQ4IDcuMTM5NTggMi4wMTgwMyA2LjQ0NzI5IDIuNTM2MyA2LjQ0MDk1TDguNTU5MTIgNi4zNjc4NUwxMC40ODY4IDAuMzg1NzQyQzEwLjY1MyAtMC4xMjg2OTggMTEuMzQ3NiAtMC4xMjg2OTggMTEuNTEzMiAwLjM4NjQ0NloiIGZpbGw9IiNEM0Q0RDUiLz4KPC9zdmc+Cg==');

$large-spacing-adjustment: -3px;
$large-height: 24px;
$large-width: 150px;
$large-image: url('data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzAiIGhlaWdodD0iNDgiIHZpZXdCb3g9IjAgMCAzMCA0OCIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPHBhdGggZD0iTTE1LjY4NDMgMjQuNTE1M0wxOC4yNTQ1IDMyLjQ5MTRMMjYuMjg0OSAzMi41ODg5QzI2Ljk3NiAzMi41OTczIDI3LjI2MiAzMy41MjA0IDI2LjcwNzggMzMuOTUyM0wyMC4yNjUgMzguOTc4N0wyMi42NTgxIDQ3LjAxNDhDMjIuODYzNyA0Ny43MDY0IDIyLjExNTUgNDguMjc2MSAyMS41NTE0IDQ3Ljg1NzJMMTUuMDAwNCA0Mi45ODczTDguNDQ4NiA0Ny44NTYzQzcuODg1NCA0OC4yNzUyIDcuMTM2MjYgNDcuNzA0NSA3LjM0MTg3IDQ3LjAxMzlMOS43MzUwMSAzOC45Nzc4TDMuMjkyMjMgMzMuOTUxM0MyLjczNzk3IDMzLjUxOTQgMy4wMjQwNCAzMi41OTY0IDMuNzE1MDcgMzIuNTg3OUwxMS43NDU1IDMyLjQ5MDVMMTQuMzE1NyAyNC41MTQzQzE0LjUzNzQgMjMuODI4NCAxNS40NjM1IDIzLjgyODQgMTUuNjg0MyAyNC41MTUzWiIgZmlsbD0iIzJEQjc4MyIvPgo8cGF0aCBkPSJNMTUuNjg0MyAwLjUxNTI2MkwxOC4yNTQ1IDguNDkxNDFMMjYuMjg0OSA4LjU4ODg3QzI2Ljk3NiA4LjU5NzMyIDI3LjI2MiA5LjUyMDM4IDI2LjcwNzggOS45NTIyOEwyMC4yNjUgMTQuOTc4N0wyMi42NTgxIDIzLjAxNDhDMjIuODYzNyAyMy43MDY0IDIyLjExNTUgMjQuMjc2MSAyMS41NTE0IDIzLjg1NzJMMTUuMDAwNCAxOC45ODczTDguNDQ4NiAyMy44NTYzQzcuODg1NCAyNC4yNzUyIDcuMTM2MjYgMjMuNzA0NSA3LjM0MTg3IDIzLjAxMzlMOS43MzUwMSAxNC45Nzc4TDMuMjkyMjMgOS45NTEzNEMyLjczNzk3IDkuNTE5NDQgMy4wMjQwNCA4LjU5NjM4IDMuNzE1MDcgOC41ODc5M0wxMS43NDU1IDguNDkwNDdMMTQuMzE1NyAwLjUxNDMyMkMxNC41Mzc0IC0wLjE3MTU5NyAxNS40NjM1IC0wLjE3MTU5NyAxNS42ODQzIDAuNTE1MjYyWiIgZmlsbD0iI0QzRDRENSIvPgo8L3N2Zz4K');

// This element will have gray stars in the background.

.root {
    position: relative;
    background-repeat: repeat-x;
    display: inline-block;
}

// This pseudo element will have yellow stars. Because of the source order it will appear over the
// top of the gray stars. The width is determined by the `data-star` value.

.root:after {
    content: '';
    background-position: bottom left;
    position: absolute;
    top: 0;
    left: 0;
    height: 100%;
}

.root[data-star='0']:after {
    width: 0%;
}

.root[data-star='0.5']:after {
    width: 10%;
}

.root[data-star='1']:after {
    width: 20%;
}

.root[data-star='1.5']:after {
    width: 30%;
}

.root[data-star='2']:after {
    width: 40%;
}

.root[data-star='2.5']:after {
    width: 50%;
}

.root[data-star='3']:after {
    width: 60%;
}

.root[data-star='3.5']:after {
    width: 70%;
}

.root[data-star='4']:after {
    width: 80%;
}

.root[data-star='4.5']:after {
    width: 90%;
}

.root[data-star='5']:after {
    width: 100%;
}

// Star sizes. We use different background images for each size for precice spacing and rendering.

.small {
    margin: 0 $small-spacing-adjustment;
    height: $small-height;
    width: $small-width;
}

.small,
.small:after {
    background-image: $small-image;
}

.medium {
    margin: 0 $medium-spacing-adjustment;
    height: $medium-height;
    width: $medium-width;
}

.medium,
.medium:after {
    background-image: $medium-image;
}

.large {
    margin: 0 $large-spacing-adjustment;
    height: $large-height;
    width: $large-width;
}

.large,
.large:after {
    background-image: $large-image;
}

// Interactive rating.

.rateInputsWrap {
    position: absolute;
    display: flex;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    z-index: 1;
}

.rateInput {
    visibility: hidden;
}

.rateLabel {
    flex: 1;
    cursor: pointer;
    // Pushes label text out of view
    text-indent: 100%;
    overflow: hidden;
}
