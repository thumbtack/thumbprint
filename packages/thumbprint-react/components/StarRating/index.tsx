import React from 'react';
import clamp from 'lodash/clamp';
import isNumber from 'lodash/isNumber';
import times from 'lodash/times';
import noop from 'lodash/noop';
import classNames from 'classnames';
import * as tokens from '@thumbtack/thumbprint-tokens';
import styles from './index.module.scss';

// Total number of stars
const MAX_NUM_STARS = 5;

// Smallest increment we render
const PRECISION = 0.5;

const StarIcon = (p: React.SVGAttributes<SVGElement>): JSX.Element => (
    <svg width="16" height="16" viewBox="0 0 16 16" xmlns="http://www.w3.org/2000/svg" {...p}>
        <path d="M8 0c-.29675 0-.54325.1998361-.62.4733224L5.8065 5.333235H.64c-.35675 0-.64.3104371-.64.670142 0 .2199454.11.4165137.27675.533399C.38 6.6066557 4.45 9.5800656 4.45 9.5800656s-1.56325 4.8000874-1.5935 4.8767541c-.02325.0698798-.03975.1467978-.03975.2264808 0 .3599563.28975.6500328.6465.6500328.13675 0 .26325-.0432349.37-.1163825L8 12.1832131s4.07325 2.9668743 4.16675 3.0337377c.10625.0731476.23325.1163825.36975.1163825.35675 0 .64675-.2933442.64675-.6500328 0-.079683-.0165-.156601-.04-.2264808-.03-.0766667-1.593-4.8767541-1.593-4.8767541s4.0695-2.9734099 4.173-3.0432896C15.89 6.4198907 16 6.2233224 16 6.0001093c0-.3564372-.27675-.6668743-.6335-.6668743H10.2L8.61975.4733224C8.54325.1998361 8.2965 0 8 0" />
    </svg>
);

interface PropTypes {
    /**
     * Number from 0-5 at increments of 0.5. Numbers between these steps will be rounded.
     */
    rating: number;
    /**
     * Number from 0-5 at increments of 1. `hoverRating` trumps `rating` with respect to star
     * highlighting.
     */
    hoverRating?: 0 | 1 | 2 | 3 | 4 | 5;
    /**
     * The size of the component when rendered
     */
    size?: 'small' | 'medium' | 'large';
    /**
     * Function that is called when a user clicks on a star. The function is supplied a single
     * parameter?: the index of the clicked star.
     */
    onStarClick?: (index: number) => void;
    /**
     * Function that is called when a user hovers over a star. The function is supplied a single
     * parameter?: the index of the hovered star.
     */
    onStarHover?: (index: number) => void;
    /**
     * Function that is called when a user mouses away from the `StarRating` component
     */
    onMouseLeave?: () => void;
}

export default function StarRating({
    rating,
    hoverRating = 0,
    size = 'small',
    onStarClick = noop,
    onStarHover = noop,
    onMouseLeave = noop,
}: PropTypes): JSX.Element {
    const clampedRating = clamp(rating, 0, MAX_NUM_STARS);
    const roundedRating = Math.round(clampedRating / PRECISION) * PRECISION;
    const ratio = (hoverRating || roundedRating) / MAX_NUM_STARS;

    // Only add title text if the stars are not clickable. Otherwise, it is confusing because the
    // title text will not match the index of the star that is being hovered on.
    // https://github.com/thumbtack/thumbprint-archive/issues/894
    const titleText =
        onStarClick !== noop ? undefined : `${roundedRating} out of ${MAX_NUM_STARS} star rating`;

    return (
        <div
            className={classNames({
                [styles.root]: true,
                [styles.rootSizeSmall]: size === 'small',
                [styles.rootSizeMedium]: size === 'medium',
                [styles.rootSizeLarge]: size === 'large',
                [styles.rootIsClickable]: onStarClick !== noop,
            })}
            title={titleText}
        >
            {/*
                Putting these first because it's positioned as absolute so that it's
                presence isn't counted
            */}
            <div
                className={`${styles.row} ${styles.rowGold}`}
                data-test-id="star-row-gold"
                onMouseLeave={onMouseLeave}
                style={{ width: isNumber(ratio) ? `${ratio * 100}%` : undefined }}
            >
                {times(MAX_NUM_STARS, index => (
                    // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
                    <div
                        key={index}
                        className={classNames({
                            [styles.starContainerSmall]: size === 'small',
                            [styles.starContainerMedium]: size === 'medium',
                            [styles.starContainerLarge]: size === 'large',
                        })}
                        onClick={(): void => onStarClick(index + 1)}
                        onMouseEnter={(): void => onStarHover(index + 1)}
                    >
                        <StarIcon
                            fill={tokens.tpColorYellow}
                            className={classNames({
                                [styles.starSVG]: true,
                                [styles.starSVGSizeSmall]: size === 'small',
                                [styles.starSVGSizeMedium]: size === 'medium',
                                [styles.starSVGSizeLarge]: size === 'large',
                            })}
                        />
                    </div>
                ))}
            </div>

            <div className={styles.row} data-test-id="star-row-gray" onMouseLeave={onMouseLeave}>
                {times(MAX_NUM_STARS, index => (
                    // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
                    <div
                        key={index}
                        className={classNames({
                            [styles.starContainerSmall]: size === 'small',
                            [styles.starContainerMedium]: size === 'medium',
                            [styles.starContainerLarge]: size === 'large',
                        })}
                        onClick={(): void => onStarClick(index + 1)}
                        onMouseEnter={(): void => {
                            onStarHover(index + 1);
                        }}
                    >
                        <StarIcon
                            fill={tokens.tpColorGray}
                            className={classNames({
                                [styles.starSVG]: true,
                                [styles.starSVGSizeSmall]: size === 'small',
                                [styles.starSVGSizeMedium]: size === 'medium',
                                [styles.starSVGSizeLarge]: size === 'large',
                            })}
                        />
                    </div>
                ))}
            </div>
        </div>
    );
}
