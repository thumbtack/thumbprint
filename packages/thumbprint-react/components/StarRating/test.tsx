import React from 'react';
import { mount, shallow } from 'enzyme';
import StarRating from './index';

test('renders 10 star SVG elements in total', () => {
    const wrapper = mount(<StarRating rating={5} />);
    expect(wrapper.find('StarIcon')).toHaveLength(10);
    expect(wrapper).toMatchSnapshot();
});

test('renders 3 star rating', () => {
    const wrapper = mount(<StarRating rating={3} />);
    expect(wrapper).toMatchSnapshot();
});

test('renders 2.5 star rating', () => {
    const wrapper = mount(<StarRating rating={2.5} />);
    expect(wrapper).toMatchSnapshot();
});

test('renders 0 star rating', () => {
    const wrapper = mount(<StarRating rating={0} />);
    expect(wrapper).toMatchSnapshot();
});

test('generates the correct title', () => {
    const wrapper = shallow(<StarRating rating={2} />);
    expect(wrapper.prop('title')).toBe('2 out of 5 star rating');
    expect(wrapper).toMatchSnapshot();
});

test('does not add title text if a function is provided', () => {
    const wrapper = shallow(<StarRating rating={2} onStarClick={jest.fn} />);
    expect(wrapper.prop('title')).toBe(undefined);
    expect(wrapper).toMatchSnapshot();
});

test('does not throw an error when clicking on a star when `onStarClick` is not provided', () => {
    const wrapper = mount(<StarRating size="large" rating={3} />);
    expect(() => {
        wrapper
            .find('[data-test-id="star-row-gold"] StarIcon')
            .first()
            .simulate('click');
    }).not.toThrow();
});

test('calls `onStarClick` function when supplied', () => {
    const onStarClick = jest.fn();
    const wrapper = mount(<StarRating rating={0} onStarClick={onStarClick} />);
    wrapper
        .find('[data-test-id="star-row-gray"] StarIcon')
        .first()
        .simulate('click');
    expect(onStarClick).toHaveBeenCalledTimes(1);
});

test('calls `onStarClick` function with correct star rating', () => {
    const onStarClick = jest.fn();
    const wrapper = mount(<StarRating rating={0} onStarClick={onStarClick} />);
    const starRatingToClick = 3;
    wrapper
        .find('[data-test-id="star-row-gray"] StarIcon')
        .at(starRatingToClick - 1) // minus one to get index
        .simulate('click');
    expect(onStarClick).toHaveBeenCalledTimes(1);
    expect(onStarClick).toHaveBeenCalledWith(starRatingToClick);
});

test('calls the `onStarHover` function with the correct star rating', () => {
    const onStarHover = jest.fn();
    const wrapper = shallow(<StarRating rating={0} onStarHover={onStarHover} />);
    const starRatingToHover = 3;
    wrapper
        .find('[data-test-id="star-row-gray"]')
        .children()
        .at(starRatingToHover - 1)
        .simulate('mouseenter');
    expect(onStarHover).toHaveBeenCalledTimes(1);
    expect(onStarHover).toHaveBeenCalledWith(starRatingToHover);
});

test('calls the `onMouseLeave` function when appropriate', () => {
    const onMouseLeave = jest.fn();
    const wrapper = shallow(<StarRating rating={0} onMouseLeave={onMouseLeave} />);
    wrapper.find('[data-test-id="star-row-gray"]').simulate('mouseleave');
    expect(onMouseLeave).toHaveBeenCalledTimes(1);
});
