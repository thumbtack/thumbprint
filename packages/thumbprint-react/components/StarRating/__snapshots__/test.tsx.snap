// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates a valid rounded rating value for the data-star attr 1`] = `
<div
  aria-label="1.5 stars out of 5 star rating"
  className="root small"
  data-star={1.5}
  onMouseLeave={[Function]}
  role="img"
/>
`;

exports[`generates the correct aria-label 1`] = `
<div
  aria-label="2 stars out of 5 star rating"
  className="root small"
  data-star={2}
  onMouseLeave={[Function]}
  role="img"
/>
`;

exports[`generates the correct aria-label with 1 star rating 1`] = `
<div
  aria-label="1 star out of 5 star rating"
  className="root small"
  data-star={1}
  onMouseLeave={[Function]}
  role="img"
/>
`;

exports[`generates the correct role attr 1`] = `
<div
  aria-label="1 star out of 5 star rating"
  className="root small"
  data-star={1}
  onMouseLeave={[Function]}
  role="img"
/>
`;

exports[`renders 0 star rating 1`] = `
<StarRating
  rating={0}
>
  <div
    aria-label="0 stars out of 5 star rating"
    className="root small"
    data-star={0}
    onMouseLeave={[Function]}
    role="img"
  />
</StarRating>
`;

exports[`renders 2.5 star rating 1`] = `
<StarRating
  rating={2.5}
>
  <div
    aria-label="2.5 stars out of 5 star rating"
    className="root small"
    data-star={2.5}
    onMouseLeave={[Function]}
    role="img"
  />
</StarRating>
`;

exports[`renders 3 star rating 1`] = `
<StarRating
  rating={3}
>
  <div
    aria-label="3 stars out of 5 star rating"
    className="root small"
    data-star={3}
    onMouseLeave={[Function]}
    role="img"
  />
</StarRating>
`;

exports[`renders 5 label elements in total 1`] = `
<StarRating
  onStarClick={[Function]}
  rating={3}
>
  <div
    aria-label="3 stars out of 5 star rating"
    className="root small"
    data-star={3}
    onMouseLeave={[Function]}
  >
    <div
      className="rateInputsWrap"
    >
      <label
        className="rateLabel"
        key="0"
        onMouseEnter={[Function]}
      >
        <input
          className="rateInput"
          name="rating"
          onClick={[Function]}
          type="radio"
          value={1}
        />
        1 star
      </label>
      <label
        className="rateLabel"
        key="1"
        onMouseEnter={[Function]}
      >
        <input
          className="rateInput"
          name="rating"
          onClick={[Function]}
          type="radio"
          value={2}
        />
        2 stars
      </label>
      <label
        className="rateLabel"
        key="2"
        onMouseEnter={[Function]}
      >
        <input
          className="rateInput"
          name="rating"
          onClick={[Function]}
          type="radio"
          value={3}
        />
        3 stars
      </label>
      <label
        className="rateLabel"
        key="3"
        onMouseEnter={[Function]}
      >
        <input
          className="rateInput"
          name="rating"
          onClick={[Function]}
          type="radio"
          value={4}
        />
        4 stars
      </label>
      <label
        className="rateLabel"
        key="4"
        onMouseEnter={[Function]}
      >
        <input
          className="rateInput"
          name="rating"
          onClick={[Function]}
          type="radio"
          value={5}
        />
        5 stars
      </label>
    </div>
  </div>
</StarRating>
`;
